mount.go

```diff
@@ -143,37 +143,78 @@ func parseMountFlagsAndOptions(optionList []string) (int, string, error) {
 	return flags, strings.Join(options, ","), nil
 }
 
-func addMounts(mounts []*pb.Storage) ([]string, error) {
+func removeMounts(mounts []string) error {
+	for _, mount := range mounts {
+		if err := syscall.Unmount(mount, 0); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type storageDriversHandler func(storage pb.Storage) (string, error)
+
+var storageDriversHandlerList = map[string]storageDriversHandler{
+	driver9pType:  storage9pDriverHandler,
+	driverBlkType: storageBlockDeviceDriverHandler,
+}
+
+func storage9pDriverHandler(storage pb.Storage) (string, error) {
+	return commonStorageHandler(storage)
+}
+
+func storageBlockDeviceDriverHandler(storage pb.Storage) (string, error) {
+	// First need to make sure the expected device shows up properly.
+	if err := waitForDevice(storage.Source); err != nil {
+		return "", err
+	}
+
+	return commonStorageHandler(storage)
+}
+
+func commonStorageHandler(storage pb.Storage) (string, error) {
+	// Mount the storage device.
+	if err := mountStorage(storage); err != nil {
+		return "", err
+	}
+
+	return storage.MountPoint, nil
+}
+
+func mountStorage(storage pb.Storage) error {
+	flags, options, err := parseMountFlagsAndOptions(storage.Options)
+	if err != nil {
+		return err
+	}
+
+	return mount(storage.Source, storage.MountPoint, storage.Fstype, flags, options)
+}
+
+func addStorages(storages []*pb.Storage) ([]string, error) {
 	var mountList []string
 
-	for _, mnt := range mounts {
-		if mnt == nil {
+	for _, storage := range storages {
+		if storage == nil {
 			continue
 		}
 
-		flags, options, err := parseMountFlagsAndOptions(mnt.Options)
-		if err != nil {
-			return nil, err
+		devHandler, ok := storageDriversHandlerList[storage.Driver]
+		if !ok {
+			return nil, grpcStatus.Errorf(codes.InvalidArgument,
+				"Unknown storage driver %q", storage.Driver)
 		}
 
-		if err := mount(mnt.Source, mnt.MountPoint, mnt.Fstype,
-			flags, options); err != nil {
+		mountPoint, err := devHandler(*storage)
+		if err != nil {
 			return nil, err
 		}
 
-		// Prepend mount point to mount list.
-		mountList = append([]string{mnt.MountPoint}, mountList...)
-	}
-
-	return mountList, nil
-}
-
-func removeMounts(mounts []string) error {
-	for _, mount := range mounts {
-		if err := syscall.Unmount(mount, 0); err != nil {
-			return err
+		if mountPoint != "" {
+			// Prepend mount point to mount list.
+			mountList = append([]string{mountPoint}, mountList...)
```

Comment:
Any reason why are you prepending here ? Why not just append with  `mountList = append(mountList, mountPoint)` ?